run_tests.py

```diff
@@ -72,7 +86,7 @@ def test_seed(seed, sim='qemu', repeat=1, retry=0):
         test_seed(seed, repeat, retry + 1)
     elif index == 3:
         print("Timeout reached.\n")
-        message = child.buffer.decode("utf-8")
+        message = safe_decode(child.buffer)
```

Comment:
As far as I understand currently message is in utf-8  and after change it will be ascii. There is no problem with that? In some places ascii was used and here it was utf-8. Is there any reason for this misalignment?